src/components/shared/Property.tsx

```diff
@@ -0,0 +1,69 @@
+import { size } from 'lodash';
+import * as React from 'react';
+import { GoToRefHandler, IArrayNode, IObjectNode, SchemaKind, SchemaNodeWithMeta } from '../../types';
+import { isCombiner, isRef } from '../../utils';
+import { inferType } from '../../utils/inferType';
+import { Types } from './Types';
+
+export interface IProperty {
+  node: SchemaNodeWithMeta;
+  onGoToRef?: GoToRefHandler;
+}
+
+export const Property: React.FunctionComponent<IProperty> = ({ node, onGoToRef }) => {
+  const type = isRef(node) ? '$ref' : isCombiner(node) ? node.combiner : node.type;
+  const subtype =
+    type === SchemaKind.Array && (node as IArrayNode).items !== undefined
+      ? inferType((node as IArrayNode).items!)
+      : undefined;
+
+  const childrenCount = React.useMemo<number | null>(
+    () => {
+      if (type === SchemaKind.Object) {
+        return size((node as IObjectNode).properties);
+      }
+
+      if (subtype === SchemaKind.Object) {
+        return size(((node as IArrayNode).items as IObjectNode).properties);
+      }
+
+      if (subtype === SchemaKind.Array) {
+        return size((node as IArrayNode).items as IArrayNode);
+      }
+
+      return null;
+    },
+    [node],
+  );
+
+  const handleGoToRef = React.useCallback<React.MouseEventHandler>(
+    () => {
+      if (onGoToRef) {
+        onGoToRef(node.$ref!, node);
+      }
+    },
+    [onGoToRef, node],
+  );
+
+  return (
```

Comment:
Similarly to my other comment all code between line 14 and `return` should/could be unit tested. What do you think?